openedx/core/djangoapps/course_groups/views.py

```diff
@@ -379,3 +390,356 @@ def debug_cohort_mgmt(request, course_key_string):
+        * invalid: Invalid emails submitted.
+
+    Adding multiple users to a cohort, send a request to:
+    POST /api/cohorts/v1/courses/{course_id}/cohorts/{cohort_id}/users
+
+    With a payload such as:
+    {
+        "users": [username1, username2, username3...]
+    }
+
+    **DELETE Response Values**
+
+        Returns a HTTP 404 Not Found response status code when:
+            * The course corresponding to the corresponding course id could not be found.
+            * The requesting user does not have staff access to the course.
+            * The cohort corresponding to the given cohort id could not be found.
+            * The user corresponding to the given username could not be found.
+        Returns a HTTP 204 No Content response status code to indicate success.
+    """
+    serializer_class = CohortUsersAPISerializer
+
+    def get(self, request, course_key_string, cohort_id, username=None):  # pylint: disable=unused-argument
+        """
+        Lists the users in a specific cohort.
+        """
+        course_key, _ = _get_course_with_access(request, course_key_string)
+
+        try:
+            cohort = cohorts.get_cohort_by_id(course_key, cohort_id)
+        except CourseUserGroup.DoesNotExist:
+            msg = 'Cohort (ID {cohort_id}) not found for {course_key_string}'.format(
+                cohort_id=cohort_id,
+                course_key_string=course_key_string
+            )
+            raise self.api_error(status.HTTP_404_NOT_FOUND, msg, 'cohort-not-found')
+        queryset = cohort.users.all()
+        page = self.paginate_queryset(queryset)
+
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        return Response(self.get_serializer(queryset, many=True).data)
+
+    # pylint: disable=unused-argument
+    def delete(self, request, course_key_string, cohort_id, username=None):
+        """
+        Removes and user from a specific cohort.
+
+        Note: It's better to use the post method to move users between cohorts.
+        """
+        if username is None:
+            raise self.api_error(status.HTTP_405_METHOD_NOT_ALLOWED,
+                                 'Missing username in path',
+                                 'missing-username')
+        course_key, _ = _get_course_with_access(request, course_key_string)
+        try:
+            api.remove_user_from_cohort(course_key, username)
+        except User.DoesNotExist:
+            raise self.api_error(status.HTTP_404_NOT_FOUND, 'User does not exist.', 'user-not-found')
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def post(self, request, course_key_string, cohort_id, username=None):
+        """
+        Add given users to the cohort.
+        """
+        course_key, _ = _get_course_with_access(request, course_key_string)
```

Comment:
Since the lines from ``_get_course_with_access`` to ``raise self.api_error`` are used in all three endpoints in this class, consolidate them into a separate method, for instance:

``course_key, cohort = self._get_course_and_cohort(request, course_key_string, cohort_id)``

Reply:
This has been addressed in the recent updates to this PR.